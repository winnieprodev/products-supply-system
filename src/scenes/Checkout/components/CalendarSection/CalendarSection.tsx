import { useCallback, useEffect, useMemo } from "react";
import { Set } from "immutable";
import Calendar, { CalendarTileProperties } from "react-calendar";

import * as Styled from "./calendarSection.styled";

import "react-calendar/dist/Calendar.css";

interface OwnProps {
  value: Date;
  onChange: (newVal: Date) => void;
  availableDates: string[];
}

const CalendarSection: React.FC<OwnProps> = ({
  value,
  onChange,
  availableDates,
}) => {
  const today = useMemo(() => new Date(), []);

  useEffect(() => {
    onChange(new Date(availableDates[0]));
  }, [availableDates]);

  const availableSet = useMemo(
    () => Set(availableDates.map((date) => new Date(date).toDateString())),
    [availableDates],
  );

  const isTileDisabled = useCallback(
    ({ date }: CalendarTileProperties) =>
      availableSet.has(date.toDateString()) === false,
    [],
  );

  return (
    <Styled.Wrapper>
      <Styled.CalendarBlock>
        <Styled.CalendarStyleOverrides>
          <Calendar
            minDate={today}
            defaultActiveStartDate={today}
            value={value}
            onClickDay={onChange}
            tileDisabled={isTileDisabled}
          />
        </Styled.CalendarStyleOverrides>
      </Styled.CalendarBlock>
      <Styled.DeliveryDateBlock>
        <Styled.DateSection>
          <Styled.DateIcon>{today.getDate()}</Styled.DateIcon>
          <Styled.DateTitle>- Today</Styled.DateTitle>
        </Styled.DateSection>
        <Styled.DateSection>
          <Styled.DateIcon $primary>{value.getDate()}</Styled.DateIcon>
          <Styled.DateTitle>- Selected date</Styled.DateTitle>
        </Styled.DateSection>
      </Styled.DeliveryDateBlock>
    </Styled.Wrapper>
  );
};

export default CalendarSection;
