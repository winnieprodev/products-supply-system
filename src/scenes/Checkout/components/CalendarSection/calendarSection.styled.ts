import styled from "styled-components";
import {
  primaryColor,
  secondaryColor,
  textOnPrimaryColor,
  textOnSecondaryColor,
} from "styles/colors";

export const Wrapper = styled.div`
  margin-top: 10px;
  margin-bottom: 30px;
  padding-left: 25px;
  display: flex;
  gap: 20px;
`;

export const CalendarBlock = styled.div`
  margin-top: 10px;
  margin-bottom: 10px;
  width: 40%;
`;

export const DeliveryDateBlock = styled.div`
  margin-left: 20px;
`;

export const DateSection = styled.div`
  margin-top: 10px;
  margin-bottom: 10px;
  display: flex;
  align-items: center;
`;

interface DateIconProps {
  $primary?: true;
}

export const DateIcon = styled.div<DateIconProps>`
  font-size: 16px;
  width: 16px;
  line-height: 16px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: ${({ $primary }) =>
    $primary ? textOnPrimaryColor : textOnSecondaryColor};
  background: ${({ $primary }) => ($primary ? primaryColor : secondaryColor)};
`;

export const DateTitle = styled.div`
  font-size: 14px;
  font-weight: 400;
  margin-left: 10px;
`;

export const CalendarStyleOverrides = styled.div`
  & .react-calendar {
    border: none;
    width: 250px;
  }

  & .react-calendar__tile--active {
    background: ${primaryColor} !important;
    color: white !important;
    border-radius: 50%;
  }

  & .react-calendar__tile:disabled {
    background-color: transparent;
    opacity: 0.4;
  }

  & .react-calendar__tile--now,
  & .react-calendar__tile--now:hover {
    background: #eeeeee;
    color: #929292;
    border-radius: 50%;
  }

  & .react-calendar__month-view__days__day--weekend {
    color: #494949;
  }

  & abbr[title] {
    text-decoration: unset !important;
  }

  & .react-calendar__month-view__weekdays {
    text-transform: unset !important;
  }

  & .react-calendar__navigation__next2-button,
  & .react-calendar__navigation__prev2-button {
    display: none;
  }

  &
    .react-calendar__navigation__prev-button
    + .react-calendar__navigation__label {
    pointer-events: none;
  }

  & .react-calendar__tile:enabled:hover,
  & .react-calendar__tile:enabled:focus {
    border-radius: 50%;
  }

  & .react-calendar__navigation button:enabled:hover,
  & .react-calendar__navigation button:enabled:focus {
    border-radius: 50%;
  }

  & .react-calendar__navigation button[disabled] {
    background-color: transparent !important;
    border-radius: 50%;
  }
`;
