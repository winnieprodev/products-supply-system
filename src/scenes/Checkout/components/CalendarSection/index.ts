export { default as CalendarSection } from "./CalendarSection";
