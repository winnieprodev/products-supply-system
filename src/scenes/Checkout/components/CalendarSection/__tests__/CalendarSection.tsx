import { render } from "@testing-library/react";

import CalendarSection from "../CalendarSection";

describe("CalendarSection", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <CalendarSection
          value={new Date("2020-10-10")}
          onChange={jest.fn()}
          availableDates={["2020-10-11", "2020-10-20"]}
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
