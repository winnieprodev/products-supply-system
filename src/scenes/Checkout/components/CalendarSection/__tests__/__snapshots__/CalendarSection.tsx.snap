// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarSection Snapshot tests it matches the snapshot 1`] = `
.c0 {
  margin-top: 10px;
  margin-bottom: 30px;
  padding-left: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 20px;
}

.c1 {
  margin-top: 10px;
  margin-bottom: 10px;
  width: 40%;
}

.c3 {
  margin-left: 20px;
}

.c4 {
  margin-top: 10px;
  margin-bottom: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 16px;
  width: 16px;
  line-height: 16px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #5F5F5F;
  background: #EEEEEE;
}

.c7 {
  font-size: 16px;
  width: 16px;
  line-height: 16px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #FFFFFF;
  background: #F25C05;
}

.c6 {
  font-size: 14px;
  font-weight: 400;
  margin-left: 10px;
}

.c2 .react-calendar {
  border: none;
  width: 250px;
}

.c2 .react-calendar__tile--active {
  background: #F25C05 !important;
  color: white !important;
  border-radius: 50%;
}

.c2 .react-calendar__tile:disabled {
  background-color: transparent;
  opacity: 0.4;
}

.c2 .react-calendar__tile--now,
.c2 .react-calendar__tile--now:hover {
  background: #eeeeee;
  color: #929292;
  border-radius: 50%;
}

.c2 .react-calendar__month-view__days__day--weekend {
  color: #494949;
}

.c2 abbr[title] {
  -webkit-text-decoration: unset !important;
  text-decoration: unset !important;
}

.c2 .react-calendar__month-view__weekdays {
  text-transform: unset !important;
}

.c2 .react-calendar__navigation__next2-button,
.c2 .react-calendar__navigation__prev2-button {
  display: none;
}

.c2 .react-calendar__navigation__prev-button + .react-calendar__navigation__label {
  pointer-events: none;
}

.c2 .react-calendar__tile:enabled:hover,
.c2 .react-calendar__tile:enabled:focus {
  border-radius: 50%;
}

.c2 .react-calendar__navigation button:enabled:hover,
.c2 .react-calendar__navigation button:enabled:focus {
  border-radius: 50%;
}

.c2 .react-calendar__navigation button[disabled] {
  background-color: transparent !important;
  border-radius: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="react-calendar"
      >
        <div
          class="react-calendar__navigation"
          style="display: flex;"
        >
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
            disabled=""
            type="button"
          >
            «
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
            disabled=""
            type="button"
          >
            ‹
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__label"
            style="flex-grow: 1;"
            type="button"
          >
            <span
              class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
            >
              April 2021
            </span>
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
            type="button"
          >
            ›
          </button>
          <button
            aria-label=""
            class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
            type="button"
          >
            »
          </button>
        </div>
        <div
          class="react-calendar__viewContainer"
        >
          <div
            class="react-calendar__month-view "
          >
            <div
              style="display: flex; align-items: flex-end;"
            >
              <div
                style="flex-grow: 1; width: 100%;"
              >
                <div
                  class="react-calendar__month-view__weekdays"
                  style="display: flex; flex-wrap: no-wrap;"
                >
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Monday"
                      title="Monday"
                    >
                      Mon
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Tuesday"
                      title="Tuesday"
                    >
                      Tue
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Wednesday"
                      title="Wednesday"
                    >
                      Wed
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Thursday"
                      title="Thursday"
                    >
                      Thu
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Friday"
                      title="Friday"
                    >
                      Fri
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Saturday"
                      title="Saturday"
                    >
                      Sat
                    </abbr>
                  </div>
                  <div
                    class="react-calendar__month-view__weekdays__weekday"
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                  >
                    <abbr
                      aria-label="Sunday"
                      title="Sunday"
                    >
                      Sun
                    </abbr>
                  </div>
                </div>
                <div
                  class="react-calendar__month-view__days"
                  style="display: flex; flex-wrap: wrap;"
                >
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 29, 2021"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 30, 2021"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="March 31, 2021"
                    >
                      31
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 1, 2021"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 2, 2021"
                    >
                      2
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 3, 2021"
                    >
                      3
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 4, 2021"
                    >
                      4
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 5, 2021"
                    >
                      5
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 6, 2021"
                    >
                      6
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 7, 2021"
                    >
                      7
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 8, 2021"
                    >
                      8
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 9, 2021"
                    >
                      9
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 10, 2021"
                    >
                      10
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 11, 2021"
                    >
                      11
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 12, 2021"
                    >
                      12
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 13, 2021"
                    >
                      13
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 14, 2021"
                    >
                      14
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 15, 2021"
                    >
                      15
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 16, 2021"
                    >
                      16
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 17, 2021"
                    >
                      17
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 18, 2021"
                    >
                      18
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 19, 2021"
                    >
                      19
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 20, 2021"
                    >
                      20
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 21, 2021"
                    >
                      21
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 22, 2021"
                    >
                      22
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 23, 2021"
                    >
                      23
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 24, 2021"
                    >
                      24
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 25, 2021"
                    >
                      25
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 26, 2021"
                    >
                      26
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 27, 2021"
                    >
                      27
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 28, 2021"
                    >
                      28
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 29, 2021"
                    >
                      29
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="April 30, 2021"
                    >
                      30
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="May 1, 2021"
                    >
                      1
                    </abbr>
                  </button>
                  <button
                    class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                    disabled=""
                    style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    type="button"
                  >
                    <abbr
                      aria-label="May 2, 2021"
                    >
                      2
                    </abbr>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c3"
  >
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        6
      </div>
      <div
        class="c6"
      >
        - Today
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c7"
      >
        10
      </div>
      <div
        class="c6"
      >
        - Selected date
      </div>
    </div>
  </div>
</div>
`;
